prov/efa/src/rdm/rxr_rma.c

```diff
@@ -219,6 +222,14 @@ ssize_t rxr_rma_readmsg(struct fid_ep *ep, const struct fi_msg_rma *msg, uint64_
 		err = -FI_EAGAIN;
 		goto out;
 	}
+
+	if (peer->is_local && rxr_ep->use_shm_for_tx) {
+		use_lower_ep_read = true;
+		// msg_clone->addr = peer->shm_fiaddr;
```

Comment:
The `fi_rma_bw` test passes when I use `msg` and `msg_clone`. I don't know how that's possible..

I have to use `msg_clone` to set the address because the input parameter `msg` is `const`. This generates a compiler warning.

Reply:
fabtests only has 1 peer that that the shm_fiaddr could be the same with efa addr. We should use shm_fiaddr for any fi_* call for shm

Reply:
I see OK

What's the best way to do this? The `msg_clone` idea which I have here generates a compiler warning. I could do a `memcpy` but that would be more expensive?

Reply:
There are `rxr_msg_construct` and `rxr_tmsg_construct` in `rxr_msg.h`. You can do similar things to have a function to construct a new fi_rma_msg